stream-chat-android-offline/src/main/java/io/getstream/chat/android/offline/channel/ChannelController.kt

```diff
@@ -804,7 +804,6 @@ public class ChannelController internal constructor(
                 deletedAt = Date()
             )
         }
-        domainImpl.repos.insertReaction(reaction)
```

Comment:
If I understand correctly, this line is redundant as reaction is saved as a part of the updated message several lines below.

Reply:
what about retries, network timeouts and the app getting killed by the OS? I mean its possible to never reach line 820 and 833 right?

Reply:
I've just rechecked and it seems to work fine as we persist the reaction on line 812:

```
val currentMessage = getMessage(reaction.messageId)?.copy() 
currentMessage?.let { // Don't understand how this can be null. If we are able to add a reaction then obviously the message should be present in `_messages` StateFlow
    it.addMyReaction(reaction, enforceUnique = enforceUnique)
    upsertMessage(it)
    domainImpl.repos.insertMessage(it) // Message is persisted here altogether with all the reactions 
}
```

Anyway, I'll ask for additional reviews on Monday. And we need to pay special attention to this functionality during regression testing before the next release.

Reply:
I don't think it retries the API call that way though. (it retries from the reactions table, not the messages table)

Reply:
Under the hood, `repos.insertMessage` persists reactions to a separate "reactions" table. If the sync status of a reaction is correct (sync_needed) then it should retry it.

```
internal abstract class MessageDao {

    @Transaction
    open suspend fun insert(messageEntities: List<MessageEntity>) {
        upsertMessageInnerEntities(messageEntities.map(MessageEntity::messageInnerEntity)) // messages table
        insertAttachments(messageEntities.flatMap(MessageEntity::attachments)) // attachments table
        insertReactions(messageEntities.flatMap { it.latestReactions + it.ownReactions }) // reactions table
    }
```

Reply:
Under the hood `insertMessage` persists reactions to the reactions table.

```
@Transaction
open suspend fun insert(messageEntities: List<MessageEntity>) {
        upsertMessageInnerEntities(messageEntities.map(MessageEntity::messageInnerEntity))
        insertAttachments(messageEntities.flatMap(MessageEntity::attachments))
        insertReactions(messageEntities.flatMap { it.latestReactions + it.ownReactions }) // persist to the reactions table with proper sync status (sync_needed)
}
```

Reply:
great. that seems good.